### PR TITLE
Support PGNs where a header is on the same line as a sequence of moves

### PR DESCRIPTION
**Pull request by dignissimus**
Resolves https://github.com/lichess-org/lila/issues/11350

### CODE REVIEW

File: src/pgn.ts [hunk 1]
```diff
@@ -445,7 +445,7 @@ export class PgnParser {
 
   private handleLine() {
     let freshLine = true;
-    let line = this.lineBuf.join('');
+    let line = this.lineBuf.join('').replace(/^\s+/, '');
```
Comment (niklasf) [inline src/pgn.ts:448]: This looks like it will break comments.

Comment (dignissimus) [reply src/pgn.ts:448]: Do you have an example for this?

I'll add it as a test, I think there should be one here.

**Comment by niklasf** [comment]
Thanks for your persistence. Getting pretty close, I think.

If move text and headers can be on the same line, I think we need to support multiple headers on the same line. Added a failing test case for that directly to your branch.

**Comment by niklasf** [comment]
Thanks again. Will do some more testing, and then make a release so we can get this on Lichess.

**Pull request merged**